numba/np/ufunc/gufunc_scheduler.cpp

```diff
@@ -11,6 +11,16 @@
 #include <stdio.h>
 #include "gufunc_scheduler.h"
 
+#ifdef _MSC_VER
+#define THREAD_LOCAL(ty) __declspec(thread) ty
+#else
+/* Non-standard C99 extension that's understood by gcc and clang */
+#define THREAD_LOCAL(ty) __thread ty
+#endif
+
+// Default 0 value means one evenly-sized chunk of work per worker thread.
+THREAD_LOCAL(uintp) parallel_chunksize = 0;
```

Comment:
Do Numba really support pre-C++11 compilers? In C++11 you can just use `thread_local`.

Also, you can make this var static or put it in anonymous namespace. And most of this file contents can be put into anonymous namespace.

Reply:
I will make it static.  In terms of the compiler, I did it this way because that is how it is done several places in Numba.  Maybe they should all change but I don't know why it is the way it is.  It could be legacy that could go away or there could still be a current reason for it.

Reply:
It's because of historically having to support Python 2.7 and older compilers. If there's a cleaner way of doing this then that's a good change to make, but perhaps this PR should focus on functionality and then in a subsequent PR we can clean the numerous implementations doing the above?

Reply:
> perhaps this PR should focus on functionality and then in a subsequent PR we can clean the numerous implementations doing the above?

Ok, sounds good